src/libutil/args.cc

```diff
@@ -622,8 +622,9 @@ std::optional<ExperimentalFeature> Command::experimentalFeature ()
     return { Xp::NixCommand };
 }
 
-MultiCommand::MultiCommand(const Commands & commands_)
+MultiCommand::MultiCommand(std::string_view commandName, const Commands & commands_)
```

Comment:
Isn't there a way to avoid passing `commandName`? Since now we pass in the same argument twice (to `MultiCommand` and `registerCommand`).

Reply:
It's indeed unfortunate. We could probably tweak the `registedCommand` logic to have it pass the command name to the constructor. Do you think that would be worth it?

Note that it was already more or less the case, since the error message of the `run` method included the method name.